pkg/devfile/adapters/common/command.go

```diff
@@ -279,3 +285,24 @@ func ValidateAndGetDebugDevfileCommands(data data.DevfileData, devfileDebugCmd s
 
 	return debugCommand, nil
 }
+
+// ValidateAndGetTestDevfileCommands validates the test command
+func ValidateAndGetTestDevfileCommands(data data.DevfileData, devfileTestCmd string) (testCommand common.DevfileCommand, err error) {
+	var emptyCommand common.DevfileCommand
+	isTestCommandValid := false
+	testCommand, testCmdErr := GetTestCommand(data, devfileTestCmd)
+	if testCmdErr == nil && !reflect.DeepEqual(emptyCommand, testCommand) {
+		isTestCommandValid = true
+		klog.V(4).Infof("Test command: %v", testCommand.Exec.Id)
+	}
+
+	if !isTestCommandValid {
+		commandErrors := ""
+		if testCmdErr != nil {
+			commandErrors += testCmdErr.Error()
+		}
+		return common.DevfileCommand{}, fmt.Errorf(commandErrors)
+	}
```

Comment:
if there is no test cmd, but i run `odo test`; it throws an empty err:
```
$ odo test

Executing devfile test command for component testm
 ✗  

```

I think the debug validation is also broken

Reply:
This is broken because of this
https://github.com/openshift/odo/blob/cd858a7c3fd897034173819d7d4bdc03dbe2e99d/pkg/devfile/adapters/common/command.go#L61-L66

test and debug type should also be included in the if statement